Bulk controller (#15)
kuzzleio/sdk-dart: doc/2/controllers/document/m-get/index.md

---
code: true
type: page
title: mGet
description: Gets multiple documents
---

# mGet

Gets multiple documents.

---

## Arguments 

```dart
Future<Map<String, dynamic>> mGet(
    String index,
    String collection,

>     List<String> ids,

Reviewer: ```suggestion
    List<String> ids
```

  )
```

| Arguments          | Type                                                    | Description                       |
| ------------------ | ------------------------------------------------------- | --------------------------------- |
| `index`            | <pre>String</pre>                                       | Index name                        |
| `collection`       | <pre>String</pre>                                       | Collection name                   |
| `ids`              | <pre>List<String></pre>                            | Document IDs                      |
---

## Return

> A `List<String>` which has a `successes` and `errors`:

Reviewer 2: ```suggestion
A `Map<String, dynamic>` which has a `successes` and `errors`:
```

Each created document is an object of the `successes` array with the following properties:

| Property     | Type                                         | Description                      |
|------------- |--------------------------------------------- |--------------------------------- |

> | `_source`    | <pre>List<String></pre> | Document content                 |

Reviewer 2: ```suggestion
| `_source`    | <pre>Map<String, dynamic></pre> | Document content                 |
```
